pwiz_tools/Skyline/SettingsUI/FullScanSettingsControl.cs

```diff
@@ -1,1110 +1,1126 @@
+                TextBox textMz = null;
+                if (analyzerTypeNew == FullScanMassAnalyzerType.qit)
+                {
+                    textMz = textRes;
+                }
+                else
+                {
+                    labelText = Resources.TransitionSettingsUI_SetAnalyzerType_Resolving_power;
+                    if (analyzerTypeNew != FullScanMassAnalyzerType.tof)
+                    {
+                        variableRes = true;
+                        textMz = textAt;
+                    }
+                }
+
+                const string resolvingPowerFormat = "#,0.####";
+                if (analyzerTypeNew == analyzerTypeCurrent && resCurrent.HasValue)
+                    textRes.Text = resCurrent.Value.ToString(resolvingPowerFormat);
+                else
+                    textRes.Text = TransitionFullScan.DEFAULT_RES_VALUES[(int)analyzerTypeNew].ToString(resolvingPowerFormat);
+
+                labelAt.Visible = variableRes;
+                textAt.Visible = variableRes;
+                textAt.Text = resMzCurrent.HasValue
+                                  ? resMzCurrent.Value.ToString(LocalizationHelper.CurrentCulture)
+                                  : TransitionFullScan.DEFAULT_RES_MZ.ToString(LocalizationHelper.CurrentCulture);
+
+                labelTh.Visible = (textMz != null);
+                if (textMz != null)
+                    labelTh.Left = textMz.Right;
+            }
+            label.Text = labelText;
+        }
+
+        private void UpdateSelectivityOption()
+        {
+            cbHighSelectivity.Enabled = IsProfileMode(comboPrecursorAnalyzerType) ||
+                                        IsProfileMode(comboProductAnalyzerType);
+            if (!cbHighSelectivity.Enabled)
+                cbHighSelectivity.Checked = false;
+        }
+
+        private bool IsProfileMode(ComboBox comboAnalyzerType)
+        {
+            if (comboAnalyzerType.Visible && comboAnalyzerType.Enabled)
+            {
+                var analyzerType = TransitionFullScan.ParseMassAnalyzer((string) comboAnalyzerType.SelectedItem);
+                return analyzerType != FullScanMassAnalyzerType.centroided && analyzerType != FullScanMassAnalyzerType.none;
+            }
+            return false;
+        }
+
+        private void InitializeUseSpectralLibraryIonMobilityUI()
+        {
+            usercontrolIonMobilityFiltering.InitializeSettings(_documentContainer);
+        }
+
+        private ImportPeptideSearchDlg.Workflow? _lastPeptideSearchWorkflow;
+        public void ModifyOptionsForImportPeptideSearchWizard(ImportPeptideSearchDlg.Workflow workflow, bool libIonMobilities)
+        {
+            var settings = _documentContainer.Document.Settings;
+
+            if (_lastPeptideSearchWorkflow == workflow)
+                return;
+            _lastPeptideSearchWorkflow = workflow;
+
+            // reset controls to default (without reconstructing them like Initialize() would do)
```

Comment:
Usually for something like this we would replace the control on the form with a new one that we construct. Is there a reason you would not want to do this here?

Reply:
I'm not actually sure this is needed after Kaipo's earlier change with MakeFullScanSettingsControl(). I'll see if it works without it. But I had been trying to call Iniitialize() instead of this and it was not working like I expected (forgot the exact problem though).